src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -292,7 +328,13 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         for (uint8_t i = 0; i < TableColorNames.size(); ++i)
         {
-            _CurrentColorTable.GetAt(i).Color(colorScheme.Table()[i]);
+            // Fill the element of "_CurrentNonBrightColorTable" with the non-bright colors (lower index on the color scheme object's table: 0-7).
+            int indexNonBrightColor = i;
+            // Fill the element of "_CurrentBrightColorTable" with the bright colors (higher index on the color scheme object's table: 8-15).
+            int indexBrightColor = i + ColorTableDivider;
+
+            _CurrentNonBrightColorTable.GetAt(i).Color(colorScheme.Table()[indexNonBrightColor]);
+            _CurrentBrightColorTable.GetAt(i).Color(colorScheme.Table()[indexBrightColor]);
         }
```

Comment:
Have you really tried this code?
I think this can't fix the error because you haven't used the whole code.

1. You didn't changed the condition of the loop which leads imo to to many iterations.

2. Please copy all my comments in the code suggestion. Now there are information that are missing.